### ISSUE TITLE
parsing error of examples in rights file

### ISSUE THREAD

**Issue by bgstack15** [Kozea/Radicale#1611]
OS: AlmaLinux 8.10
Problem: when using the default `/etc/radicale/rights` file with some examples uncommented, a parsing error occurs which then stops the parsing entirely. The program does not recover and continue with any other rules.

From journalctl -f -u radicale.service:
```
Nov 01 11:01:13 server3.ipa.internal.com radicale[8675]: [8675/Thread-2] [INFO] Successful login: 'bgstack15@IPA.INTERNAL.COM'
Nov 01 11:01:13 server3.ipa.internal.com radicale[8675]: [8675/Thread-2] [ERROR] An exception occurred during REPORT request on '/bgstack15/': Error in section 'read-domain-principal' of rights file '/etc/radicale/rights': Replacement index 0 out of range for positional args tuple
```

Here is the packages related to python and radicale I just updated:

```
    Upgrade       python39-3.9.20-1.module_el8.10.0+3902+1690be06.x86_64      @appstream
    Upgraded      python39-3.9.19-7.module_el8.10.0+3885+d986a391.x86_64      @@System
    Upgrade       python39-libs-3.9.20-1.module_el8.10.0+3902+1690be06.x86_64 @appstream
    Upgraded      python39-libs-3.9.19-7.module_el8.10.0+3885+d986a391.x86_64 @@System
    Upgrade       python3-perf-6.11.6-1.el8.elrepo.x86_64                     @elrepo-kernel
    Upgraded      python3-perf-6.11.2-1.el8.elrepo.x86_64                     @@System
    Upgrade       python3-copr-2.0-1.el8.noarch                               @epel
    Upgraded      python3-copr-1.132-1.el8.noarch                             @@System
    Upgrade       python3-koji-1.35.1-1.el8.noarch                            @epel
    Upgraded      python3-koji-1.35.0-1.el8.noarch                            @@System
    Upgrade       python39-radicale3-3.3.0-1.el8.noarch                       @epel
    Upgraded      python39-radicale3-3.2.3-1.el8.noarch                       @@System
    Upgrade       radicale3-3.3.0-1.el8.noarch                                @epel
    Upgraded      radicale3-3.2.3-1.el8.noarch                                @@System
   3 warning: /etc/radicale/config created as /etc/radicale/config.rpmnew
   4 warning: /etc/radicale/rights created as /etc/radicale/rights.rpmnew
```

The offending section appears to be:

```
# Allow reading the domain collection
[read-domain-principal]
user: .+@([^@]+)
collection: {0}
permissions: R
```

I suppose the problem would be in how capture groups are not accessible by the parser for the "collection" attribute on the permission directive, but I cannot figure how to re-enable that, or otherwise accomplish something similar. A named capture group did NOT work. E.g.:

```
# Allow reading the domain collection
[read-domain-principal]
user: .+@(?P<usern>[^@]+)
collection: {usern}
permissions: R
```

My end goal is to use a rule like so, as of radicale 3.2.3 parsing:

```
[principal-domain]
user: (.+)@IPA.EXAMPLE.COM
collection: {0}
permissions: RW
```

**Comment by pbiering** [comment]
@petervarkoly can you please investigate

**Comment by pbiering** [comment]
@bgstack15 are you using the new LDAP authentication plugin available since 3.3.0 ?

**Comment by bgstack15** [comment]
No, sorry, I hadn't realized there was an ldap auth mechanism. For some reason when I was trying to peruse the changelog I completely missed that. I have my reverse proxy handle authentication, so that radicale uses:
```
[auth]
type = http_x_remote_user
```
My header then comes in with "bgstack15@IPA.EXAMPLE.COM" and then I've always had to patch app/__init__.py to `user = user.split("@")[0]` at the right spot. I'm not proud, but it works. So technically I wasn't even using the rule demonstrated, but was trying to troubleshoot the error and was able to derive that the example rule does not work.

**Comment by pbiering** [comment]
aren't you able to strip the domain on reverse proxy level before calling _radicale_?

**Comment by bgstack15** [comment]
Yes, you inspired me to fix it where the problem lies. I finally learned how to do it!

My environment probably will not have multiple domains for gssapi auth, so in apache httpd I now strip off the [strip off the domain name](https://stackoverflow.com/questions/44528316/apache-httpd-2-2-how-to-transform-value-of-a-variable-to-remove-part-of-the-st/77623243#77623243), for my /radicale/ endpoint, which allows me to revert to official `app/__init__.py` without modifications.

Part of my apache httpd.conf:
```
RewriteEngine On
RewriteRule ^/radicale$ /radicale/ [R,L]
<Location "/radicale/">
   ProxyPreserveHost On
   Include conf.d/auth-gssapi.cnf
   Require valid-user
   AuthName "GSSAPI protected"
   ProxyPass        http://localhost:5232/ retry=20 connectiontimeout=300 timeout=300
   ProxyPassReverse http://localhost:5232/
   RequestHeader    set X-Script-Name /radicale
+   RequestHeader    edit X_REMOTE_USER "^(.*)@.*" "$1"
</Location>
```
Thank you for inspiring me to do that!

One reason I have this setup is for Kerberos (which goes to the same ldap directory I would use if I used the ldap auth plugin, of course).

**Comment by pbiering** [comment]
@petervarkoly can you please investigate why provided example (afair added by you) has an issue

**Comment by pbiering** [comment]
Created a Wiki page with hints https://github.com/Kozea/Radicale/wiki/Reverse-Proxy-Hints

**Comment by bgstack15** [comment]
This page might be useful to update as well: https://radicale.org/v3.html#reverse-proxy, or have it link to your wiki page. And if you care about my entire auth-gssapi.cnf file that is included in my above snippet, here it is. I realize it might be out of scope for what you want to provide.
```
# File: /etc/httpd/conf.d/auth.cnf
# Reference:
#    https://github.com/gssapi/mod_auth_gssapi
AuthType GSSAPI
GssapiUseSessions On
Session On
SessionCookieName s1_session path=/;
GssapiCredStore keytab:/etc/httpd/keytab
GssapiCredStore ccache:/etc/httpd/krb5.cache               
SessionHeader S1SESSION
GssapiSessionKey file:/etc/httpd/gssapisession.key
GssapiImpersonate On
GssapiDelegCcacheDir /run/httpd/ccache
GssapiDelegCcachePerms mode:0660 gid:apache
GssapiUseS4U2Proxy On
GssapiAllowedMech krb5
GssapiBasicAuth On
GssapiBasicAuthMech krb5
GssapiLocalName On
GssapiNameAttributes json
AuthBasicProvider ldap
AuthLDAPGroupAttribute member
AuthLDAPSubGroupClass group
AuthLDAPGroupAttributeIsDN On
AuthLDAPURL "ldaps://dns1.ipa.internal.com:636 dns2.ipa.internal.com:636/cn=users,cn=accounts,dc=ipa,dc=internal,dc=com?uid,memberof,gecos?sub?(objectClass=person)"
#GSS_NAME returns username@IPA.EXAMPLE.COM which merely needs additional rules in /etc/radicale/rights
RequestHeader set X_REMOTE_USER "%{GSS_NAME}e"
# Does not work
#RequestHeader set X_GROUPS "%{AUTHENTICATE_memberOf}e"
# mostly useless values
#RequestHeader set X_REMOTE_GSS "%{GSS_NAME_ATTRS_JSON}e"
```
So after this RequestHeader set goes the RequestHeader edit command that strips the domain.

**Comment by TownCube** [comment]
This also broke for me even though I was using native Radicale `.htpasswd` authentication. A proposed fix is in #1646
